include/OpenSpaceToolkit/Astrodynamics/NumericalSolver.hpp

```diff
@@ -0,0 +1,261 @@
+                                                                                const   Instant&                    anEndInstant,
+                                                                                const   SystemOfEquationsWrapper&   aSystemOfEquations ) const ;
+
+        /// @brief              Perform numerical integration for a certain duration
+        ///
+        /// @code
+        ///                     StateVector stateVector = numericalsolver.integrateStateFromInstantToInstant(stateVector, instant, otherInstant, SystemofEquations) ;
+        /// @endcode
+        /// @param              [in] anInitialStateVector An initial n-dimensional state vector to begin integrating at
+        /// @param              [in] anIntegrationDuration A duration over which to integration
+        /// @param              [in] aSystemOfEquations An std::function wrapper with a particular signature that boost::odeint accepts to perform numerical integration
+        /// @return             std::vector<double>
+
+        StateVector             integrateStateForDuration                   (   const   StateVector&                anInitialStateVector,
+                                                                                const   Duration&                   anIntegrationDuration,
+                                                                                const   SystemOfEquationsWrapper&   aSystemOfEquations ) const ;
+
+        /// @brief              Get string from the integration stepper type
+        ///
+        /// @code
+        ///                     NumericalSolver::StringFromIntegrationStepperType() ;
+        /// @endcode
+        ///
+        /// @return             IntegrationStepperType
+
+        static String           StringFromIntegrationStepperType            (   const   NumericalSolver::IntegrationStepperType&  aIntegrationStepperType            ) ;
+
+        /// @brief              Get string from the integration log type
+        ///
+        /// @code
+        ///                     NumericalSolver::StringFromIntegrationLogType() ;
+        /// @endcode
+        ///
+        /// @return             IntegrationLogType
+
+        static String           StringFromIntegrationLogType                (   const   NumericalSolver::IntegrationLogType&  aIntegrationLogType                    ) ;
+
+    private:
+
+        NumericalSolver::IntegrationLogType integrationLogType_ ;
+        NumericalSolver::IntegrationStepperType integrationStepperType_ ;
+        Real timeStep_ ;
+        Real relativeTolerance_ ;
+        Real absoluteTolerance_ ;
+
+        static void             NumericalIntegrationLogger                  (   const   StateVector&                x,
+                                                                                const   double                      t                                           ) ;
```

Comment:
`const double`, not `const double&` for a reason?

Reply:
fixed